src/gossip/service.zig

```diff
@@ -1804,16 +1804,37 @@ pub const GossipService = struct {
         self: *Self,
         /// the current time to insert the values with
         now: u64,
-    ) void {
-        var push_msg_queue, var push_msg_queue_lock = self.push_msg_queue_mux.writeWithLock();
+    ) !void {
+        const push_msg_queue, var push_msg_queue_lock = self.push_msg_queue_mux.writeWithLock();
         defer push_msg_queue_lock.unlock();
 
-        var gossip_table, var gossip_table_lock = self.gossip_table_rw.writeWithLock();
+        const gossip_table, var gossip_table_lock = self.gossip_table_rw.writeWithLock();
         defer gossip_table_lock.unlock();
 
-        while (push_msg_queue.popOrNull()) |gossip_value| {
-            _ = gossip_table.insert(gossip_value, now) catch {};
-        }
+        // number of items consumed, starting from the beginning of the queue
+        const consumed_item_count, const maybe_err = for (push_msg_queue.items, 0..) |gossip_value, i| {
+            const result = gossip_table.insert(gossip_value, now) catch |err| break .{ i, err };
+            switch (result) {
+                // good and expected
+                .InsertedNewEntry => {},
+                .OverwroteExistingEntry => {},
+
+                // concerning
+                .IgnoredOldValue => self.logger.warn().logf("Ignored old value: {}", .{gossip_value}), // TODO: should this be a warning, or just info?
```

Comment:
i think warn should be good 

maybe we add more context: "DrainPushMessages: Ignored old value: {}"

Reply:
Same for the prong immediately below this one?

Reply:
ye

Reply:
7c8e445dc3d70a2fa74a3e428214e30566b7bfa4